File: tests/resources/test_codegen/test_default_code_generators_and_templates/python/expected_echo_generated_code.py
#!/usr/bin/env python3
import sys

MOD = 123  # type: int
YES = "yes"  # type: str
NO = "NO"  # type: str


def solve(N: int, M: int, H: "List[List[str]]", A: "List[int]", B: "List[float]", Q: int, X: "List[int]"):
    print(N, M)
    assert len(H) == N - 1
    for i in range(N - 1):
        assert len(H[i]) == M - 2
        print(*H[i])
    assert len(A) == N - 1
    assert len(B) == N - 1
    for i in range(N - 1):
        print(A[i], B[i])
    print(Q)
    assert len(X) == M + Q
    for i in range(M + Q):
        print(X[i])

    print(YES)
    print(NO)
    print(MOD)


def main():
    def iterate_tokens():
        for line in sys.stdin:
            for word in line.split():
                yield word
    tokens = iterate_tokens()
    N = int(next(tokens))  # type: int
    M = int(next(tokens))  # type: int
    H = [[next(tokens) for _ in range(M - 1 - 2 + 1)] for _ in range(N - 2 + 1)]  # type: "List[List[str]]"
    A = [int()] * (N - 2 + 1)  # type: "List[int]"
    B = [float()] * (N - 2 + 1)  # type: "List[float]"
    for i in range(N - 2 + 1):
        A[i] = int(next(tokens))
        B[i] = float(next(tokens))
    Q = int(next(tokens))  # type: int
    X = [int(next(tokens)) for _ in range(M + Q)]  # type: "List[int]"
    solve(N, M, H, A, B, Q, X)

if __name__ == '__main__':
    main()

File: tests/resources/test_codegen/test_default_code_generators_and_templates/julia/expected_echo_generated_code.jl
#!/usr/bin/env julia
const MOD = 123
const YES = "yes"
const NO = "NO"

function solve(N::Int, M::Int, H::Matrix{String}, A::Vector{Int}, B::Vector{Float64}, Q::Int, X::Vector{Int})
    println("$N $M")
    @assert size(H) == (N-1, M-2)
    for i in 1:size(H,1)
        println(join(H[i, :], " "))
    end
    @assert length(A) == N - 1
    @assert length(B) == N - 1
    for (a, b) in zip(A, B)
        println("$a $b")
    end
    println(Q)
    @assert length(X) == M + Q
    foreach(println, X)

    println(YES)
    println(NO)
    println(MOD)
end


function main()
    tokens = Channel{String}(32)
    Task() do
        for line in eachline(@static VERSION < v"0.6" ? STDIN : stdin)
            for token in split(chomp(line))
                put!(tokens, token)
            end
        end
        close(tokens)
    end |> schedule
    N = parse(Int, take!(tokens))
    M = parse(Int, take!(tokens))
    H = similar(Matrix{String}, N-2+1, M-1-2+1)
    for i in 1:(N-2+1)
        for j in 1:(M-1-2+1)
            H[i,j] = take!(tokens)
        end
    end
    A = similar(Vector{Int}, N-2+1)
    B = similar(Vector{Float64}, N-2+1)
    for i in 1:(N-2+1)
        A[i] = parse(Int, take!(tokens))
        B[i] = parse(Float64, take!(tokens))
    end
    Q = parse(Int, take!(tokens))
    X = similar(Vector{Int}, M+Q)
    for i in 1:(M+Q)
        X[i] = parse(Int, take!(tokens))
    end
    solve(N, M, H, A, B, Q, X)
end

isempty(ARGS) && main()
